tests/components/sonos/test_media_player.py

```diff
@@ -768,3 +774,41 @@ async def test_service_snapshot_restore(
             blocking=True,
         )
     assert mock_restore.call_count == 2
+
+
+async def test_volume(
+    hass: HomeAssistant,
+    soco: MockSoCo,
+    async_autosetup_sonos,
+) -> None:
+    """Test the media player volume services."""
+    initial_volume = soco.volume
+
+    await hass.services.async_call(
+        MP_DOMAIN,
+        SERVICE_VOLUME_UP,
+        {
+            "entity_id": "media_player.zone_a",
+        },
+        blocking=True,
+    )
+    assert soco.volume == initial_volume + VOLUME_INCREMENT
```

Comment:
Integration tests should only validate towards state machine and other "official" apis